src/symbolize/gimli/parse_running_mmaps_unix.rs

```diff
@@ -53,19 +53,34 @@ pub(super) struct MapsEntry {
     pathname: OsString,
 }
 
-pub(super) fn parse_maps() -> Result<Vec<MapsEntry>, &'static str> {
-    let mut v = Vec::new();
+fn concat_str(a: &str, b: &str) -> String {
+    let mut e = String::from(a);
+    e += b;
+    e
+}
+
+fn read_file(file: &str) -> Result<String, String> {
     let mut proc_self_maps =
-        File::open("/proc/self/maps").map_err(|_| "Couldn't open /proc/self/maps")?;
+        File::open(file).map_err(|_| concat_str("Couldn't open file: ", file))?;
     let mut buf = String::new();
     let _bytes_read = proc_self_maps
         .read_to_string(&mut buf)
-        .map_err(|_| "Couldn't read /proc/self/maps")?;
-    for line in buf.lines() {
-        v.push(line.parse()?);
-    }
+        .map_err(|_| concat_str("Couldn't read file: ", file))?;
+    Ok(buf)
+}
```

Comment:
This machinery is a reimplementation of [`fs::read_to_string`](https://doc.rust-lang.org/std/fs/fn.read_to_string.html). You can import `fs` by changing
```diff
- use super::mystd::fs::File;
+ use super::mystd::fs::{self, File};
```

If you must, you can use `map_err` for the Err variant.